casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -300,28 +300,69 @@ object BlockAPI {
 
     def casperResponse(implicit casper: MultiParentCasper[F]): F[ApiErr[A]] =
       for {
-        dag      <- MultiParentCasper[F].blockDag
-        depth    <- maybeDepth.fold(dag.topoSort(0L).map(_.length - 1))(_.pure[F])
-        topoSort <- dag.topoSortTail(depth)
-        result   <- doIt(casper, topoSort)
+        dag               <- MultiParentCasper[F].blockDag
+        latestBlockNumber <- dag.latestBlockNumber
+        topoSort          <- dag.topoSort((latestBlockNumber - depth).toLong, none)
```

Comment:
If you like, you can create `topoSortTail` extension method on BlockDagRepresentation as a higher order part of Dag algebra. It can be together with the interface or in the casper project.
Boolean extensions seems like a good example with companion object.
https://github.com/rchain/rchain/blob/5670b00bc/shared/src/main/scala/coop/rchain/catscontrib/BooleanF.scala

Reply:
To be honest, I don't really know how to do it. I need to understand that.

Reply:
It's totally ok. We can play with extensions together.